webapp/src/utils/date/dateUtils.ts

```diff
@@ -0,0 +1,217 @@
+  return locale === "fr" ? fr : enUS;
+}
+
+/**
+ * Parse a flexible date string using multiple formats
+ *
+ * @param dateStr - The date string to parse
+ * @param locale - The locale to use for parsing
+ * @returns A valid Date object or null if parsing failed
+ */
+export function parseFlexibleDate(dateStr: string, locale: SupportedLocale = "en"): Date | null {
+  const dateFnsLocale = getDateFnsLocale(locale);
+
+  // Try each format until we find one that works
+  const formats = Object.values(DATE_FORMATS);
+
+  for (const formatStr of formats) {
+    try {
+      const parsedDate = parse(dateStr, formatStr, new Date(), { locale: dateFnsLocale });
+      if (isValid(parsedDate)) {
+        return parsedDate;
+      }
+    } catch {
+      // Continue to next format
+    }
+  }
+
+  // Special handling for formats without year
+  if (!dateStr.includes("20")) {
+    const currentYear = new Date().getFullYear();
+
+    // Try common patterns and add current year
+    const patternsWithYear = [
+      // "Mon 1 Jan 00:00" -> "Mon 1 Jan 2024 00:00"
+      dateStr.replace(/^(\w{3}\s+\d{1,2}\s+\w{3})(\s+\d{2}:\d{2})$/, `$1 ${currentYear}$2`),
+      // "1 Jan 00:00" -> "1 Jan 2024 00:00"
+      dateStr.replace(/^(\d{1,2}\s+\w{3})(\s+\d{2}:\d{2})$/, `$1 ${currentYear}$2`),
+      // Generic pattern
+      dateStr.replace(/(\d{1,2}\s+\w{3})(\s+\d{2}:\d{2})?/, `$1 ${currentYear}$2`),
+    ];
+
+    for (const dateWithYear of patternsWithYear) {
+      for (const formatStr of formats) {
+        try {
+          const parsedDate = parse(dateWithYear, formatStr, new Date(), { locale: dateFnsLocale });
+          if (isValid(parsedDate)) {
+            return parsedDate;
+          }
+        } catch {
+          // Continue
+        }
+      }
+    }
+  }
+
+  return null;
+}
+
+/**
+ * Find weekday index from a string (Monday = 1, Sunday = 7)
+ *
+ * @param dateStr - The string to search in
+ * @param locale - The locale to use
+ * @returns Weekday index (1-7) or null if not found
+ */
+export function findWeekdayIndex(dateStr: string, locale: SupportedLocale = "en"): number | null {
+  const lowerStr = dateStr.toLowerCase();
+  const weekdayNames = WEEKDAY_NAMES[locale];
+
+  // Check abbreviations first (more specific)
+  for (let i = 0; i < weekdayNames.abbreviation.length; i++) {
+    if (lowerStr.startsWith(weekdayNames.abbreviation[i])) {
+      // Convert from 0-based Sunday=0 to 1-based Monday=1
+      return i === 0 ? 7 : i;
+    }
+  }
+
+  // Check short names
+  for (let i = 0; i < weekdayNames.short.length; i++) {
+    if (lowerStr.startsWith(weekdayNames.short[i].toLowerCase())) {
+      return i === 0 ? 7 : i;
+    }
+  }
+
+  return null;
+}
+
+/**
+ * Find month index from a string (1-based)
+ *
+ * @param dateStr - The string to search in
+ * @param locale - The locale to use
+ * @returns Month index (1-12) or null if not found
+ */
+export function findMonthIndex(dateStr: string, locale: SupportedLocale = "en"): number | null {
```

Comment:
I think it's better to add a `parse` method in `TIME_FREQUENCY_CONFIG`, and works with `Date` object (`date.getMonth()`).

Reply:
And if possible use a CustomCell to handle Dates